### PR TITLE
Prepare 2.0 release

### PR DESCRIPTION
**Pull request by gchtr**
This pull request prepares the docs for the 2.x release. It will be used in combination with https://github.com/timber/timber/pull/2829.

- Upgrades Eleventy.js from 1.x to 2.x
- Makes /v2/ the default version by adding a redirect from `https://timber.github.io/docs/` to `https://timber.github.io/docs/v2/`.
- Adds an index file for 1.x in `https://timber.github.io/docs/v1/`.
- Adds a version note that always displays which version you are currently reading.

## Testing

This version of the docs can be tested when

- The [`2.x-prepare-release`](https://github.com/timber/timber/pull/2829) branch is checked out in a **timber** directory right next to the **docs** directory.
- You use PHP 7.4 for the local docs testing site.

Then, from the **docs** folder, run `npm run build`. You then use `npm run serve` to check out the docs from the local web server address displayed in your terminal.

### CODE REVIEW

**Review by Levdbas — COMMENTED** [review]
Great work @gchtr ! I found a few small things one reviewing the docs. Some are not directly connected to this PR though but are small improvements.

- [x] https://timber.github.io/docs/v2/guides/posts/#extending-timber\post breaks anchor
- [x] .toc a might be better as a inline-block, so the link does not stretch over the full width of the page
- [ ] pages link http://localhost:8080/docs/guides/custom-page-templates/ (v1) are not under the v1 slug. I think this might be more consistent.

**Comment by gchtr** [comment]
@Levdbas Thanks a lot for your review!

> * [x]  [https://timber.github.io/docs/v2/guides/posts/#extending-timber\post](https://timber.github.io/docs/v2/guides/posts/#extending-timber%5Cpost) breaks anchor

Fixed in https://github.com/timber/docs/pull/80/commits/e3e7f7bf4dcbaf04427ec620167d9a2b141fd0c5.

> * [x]  .toc a might be better as a inline-block, so the link does not stretch over the full width of the page

Fixed in https://github.com/timber/docs/pull/80/commits/3ccbb96b1ee7122595fc76b398d313efe6fca177

> * [ ]  pages link [http://localhost:8080/docs/guides/custom-page-templates](http://localhost:8080/docs/guides/custom-page-templates/) (v1) are not under the v1 slug. I think this might be more consistent.

Yes, I thought about that, too. But then we would have to create redirects for all v1 pages from `/` to `/v1/` to not break a whole lot links out there. I’m not sure whether it’s worth it.

**Comment by Levdbas** [comment]
Great work!

> Yes, I thought about that, too. But then we would have to create redirects for all v1 pages from / to /v1/ to not break a whole lot links out there. I’m not sure whether it’s worth it.

Hmm, on the other hand. It seems that the v1 docs have more "weight" currently then the soon to be new major version.
I really like the bootstrap idea now I think of it: https://getbootstrap.com/ The homepage points to the latest docs at https://getbootstrap.com/docs/5.3/.

What if we put all v1 docs under the /v1/ slug and then redirect all traffic to the new /v2/ as a permanent redirect.  Some links will then give a 404 since we changed slugs etc.

From there it might be easier to release a v3 in the future as well. I say, let's take the hit now if any and profit in the future.

let me know what you think!

**Comment by gchtr** [comment]
Okay, you have a point here as well. It sure is cleaner for future versions and I didn’t consider the possible SEO hit yet. I guess for that to work, we would need to work through the following todos:

- [x] Keep v1 pages in /v1/ folder @gchtr
- [ ] Update all links in in the Markdown files in /docs/v1/ in the https://github.com/timber/timber repo @Levdbas 
- [x] Create redirects for the current links on the main level to their corresponding pages on v2. I think we can’t just redirect all traffic to their corresponding v2 page automatically, because this is a static site. And GitHub pages doesn’t have an option like that. @gchtr 
- [x] Create a custom 404 page: https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-custom-404-page-for-your-github-pages-site (I just read now we can do that, because currently the 404 is not so nice: https://timber.github.io/docs/404) @gchtr

**Comment by Levdbas** [comment]
Gotcha! I will prepare a PR for the 2nd todo. That PR should be based on the 2.x branch right but for the /v1 docs.

**Comment by gchtr** [comment]
Thanks for taking over the second task!

> That PR should be based on the 2.x branch right but for the /v1 docs.

Yes!

**Comment by Levdbas** [comment]
Submitted at https://github.com/timber/timber/pull/2836

**Comment by gchtr** [comment]
@Levdbas Huh, I think I figured it all out. There’s a way in Eleventy to create pages from arrays, so the redirects should now be handled without us having to define them manually. I even managed to make it work so that if we release v3, all non-versioned links automatically point to v3.

I’ll have a look at your PR soon.

**Comment by gchtr** [comment]
I added a fix for the archive pages in https://github.com/timber/docs/pull/80/commits/11b2d7bb758740a72decbf76183882ee237615ce. Maybe you can test it when you pull the latest `timber-2.x` from the docs repository.

CleanShot 2023-11-07 at 17 38 33

Except for the section titles in the nav bar when accessing an archive page, everything seems to work fine.

File: _includes/sidebar.njk [hunk 1]
```diff
@@ -12,19 +12,19 @@
...
+    {% set collection_slug = ['v', version]|join %}
+
+    {% if collections[collection_slug] %}
+        <nav id="nav-main" class="navMain">
+            <ul class="navMain-sections">
+                {% for section in collections[collection_slug] %}
+                    {% set item = nav[section.slug] %}
 
                     <li class="navMain-section">
                         <h3 class="navMain-title {% if item.name === sectionTitle %}{# Marking the current section for Algolia DocSearch. #}navMain-sectionTitle{% endif %}">{{ item.name }}</h3>
```
Comment (Levdbas) [inline _includes/sidebar.njk]:
```suggestion
                        <h3 class="navMain-title {% if item.name === sectionTitle %}{# Marking the current section for Algolia DocSearch. #}navMain-sectionTitle{% endif %}">{{ section.title }}</h3>
```

Fixes the missing section title on archive pages.

**Review by Levdbas — CHANGES_REQUESTED** [review]
Suggested a fix for the missing section titles, other that that we are good to go from what I can see!

File: _includes/sidebar.njk (hunk 1 above)
Comment (gchtr) [reply _includes/sidebar.njk]: 🙈 Just what I was looking for! Thanks a lot, @Levdbas!

**Pull request merged**